### ISSUE TITLE
setImmediate usage prevents deploying to Edge Runtime

### ISSUE THREAD

**Issue by maciej-trebacz** [BuilderIO/builder#2572]
**Describe the bug**
I'm trying to deploy a NextJS app that uses Builder SDK to Edge Runtime. Unfortunately when I configure the page route to use the `edge` runtime I'm encountering the following error:

```
- error Error: A Node.js API is used (setImmediate) which is not supported in the Edge Runtime.
Learn more: https://nextjs.org/docs/api-reference/edge-runtime
    at Page (./app/[[...page]]/page.tsx:17:6)
```

It looks like `setImmediate` usage in [nextTick function](https://github.com/BuilderIO/builder/blob/03f4fdc1c97706d3491de329e563da2b1229a89b/packages/core/src/functions/next-tick.function.ts#L10) is the culprit here, it's a non standard API that's not implemented in the Edge Runtime. Since it's functionally almost the same as `setTimeout(callback, 0)` or creating a new Promise that's resolved immediately could we remove it from the SDK, so that using the Edge Runtime is possible?

**To Reproduce**
1. Create a new NextJS app with App Router
2. Configure the `[[...page]]/page.tsx` route with the following line: `export const runtime = 'edge';`
3. Run the development server
4. Navigate to localhost:3000 and observe the error in the console

**Comment by js-easy-school** [comment]
Here's an example of replacing setImmediate with setTimeout:

// Replace this code
setImmediate(() => {
  // Your code
});

// With this code
setTimeout(() => {
  // Your code
}, 0);

After making these changes, you should rebuild and deploy your NextJS application using the Builder SDK for the Edge Runtime. This should resolve the error related to using setImmediate in the Edge Runtime environment.

**Comment by maciej-trebacz** [comment]
It is not my code that contains the setImmediate call, it’s the Builder SDK code so I cannot replace the call myself.

**Comment by teleaziz** [comment]
Thanks @maciej-trebacz for reporting this issue and providing the fix for it, we'll release it asap in next version

**Comment by teleaziz** [comment]
Fix is released on `3.0.14` please update your `@builder.io/react` version to latest to get the fix
